e2e_test/backfill/backfill_progress/create_materialized_view_snapshot.slt

```diff
@@ -0,0 +1,98 @@
+statement ok
+set backfill_rate_limit=100;
```

Comment:
For snapshot backfill, we have to create a separate test file because of some differences:
1. The backfill rate limit can be set, but can't be altered.
2. The rate limit doesn't seem to be applied strictly, I see the progress to be much higher than in create_materialized_view.slt.

For these reasons I'm applying a higher backfill rate limit, so the backfill can complete in a reasonable amount of time. I also tweak the validation steps to validate earlier, in case the backfill finishes ahead of time due to (2).